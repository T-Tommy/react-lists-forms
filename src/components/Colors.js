import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import Color from './Color';

export default class Colors extends PureComponent {
  static propTypes = {
    colors: PropTypes.array.isRequired
  }

  render() {
    const { colors } = this.props;
    return (
      <ul>
        {
          colors.map(color =>
            <li key={ color.name }>
              <Color color={ color } />
            </li>
          )
        }
      </ul>
    );
  }
}
