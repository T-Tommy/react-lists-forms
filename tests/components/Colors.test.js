import React from 'react';
import { shallow } from 'enzyme';
import Colors from '../../src/components/Colors';

describe('Colors component', () => {
  it('renders Colors', () => {
    const colors = [
      {
        name: 'Blue',
        hex: '#FF00AA'
      },
      {
        name: 'Blaci',
        hex: '#424242'
      }
    ];

    const wrapper = shallow(<Colors colors={ colors } />);
    expect(wrapper).toMatchSnapshot();
  });
});
