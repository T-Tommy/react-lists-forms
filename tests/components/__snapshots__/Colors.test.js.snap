// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Colors component renders Colors 1`] = `
<ul>
  <li
    key="Blue"
  >
    <Color
      color={
        Object {
          "hex": "#FF00AA",
          "name": "Blue",
        }
      }
    />
  </li>
  <li
    key="Blaci"
  >
    <Color
      color={
        Object {
          "hex": "#424242",
          "name": "Blaci",
        }
      }
    />
  </li>
</ul>
`;
